src/main/java/com/ericsson/esj/JenkinsManager.java

```diff
@@ -0,0 +1,259 @@
+        String authString = String.join(":", username, password);
+
+        this.jenkinsBaseUrl = jenkinsBaseUrl;
+        this.encoding = DatatypeConverter.printBase64Binary(authString.getBytes("utf-8"));
+        this.crumb = getCrumb();
+    }
+
+    /**
+     * Constructor, takes jenkins protocol, host, port, username and password.
+     * @param protocol
+     * @param host
+     * @param port
+     * @param username
+     * @param password   **  Jenkins password or API-token
+     * @throws UnsupportedEncodingException
+     * @throws URISyntaxException
+     * @throws JSONException
+     */
+    public JenkinsManager(String protocol, String host, int port, String username, String password)
+            throws UnsupportedEncodingException, URISyntaxException, JSONException {
+        String authString = String.join(":", username, password);
+        this.jenkinsBaseUrl = String.format("%s://%s:%d",protocol, host, port);
+        this.encoding = DatatypeConverter.printBase64Binary(authString.getBytes("utf-8"));
+        this.crumb = getCrumb();
+    }
+
+    /**
+     * Takes a token and script and returns XML data for jenkins with job_token and
+     * script added.
+     *
+     * @param job_token
+     * @param script
+     * @return
+     * @throws IOException
+     */
+    public String getXmlJobData(String job_token, String script) throws IOException {
+        String jobData = getStringFromFile(JENKINS_JOB_TEMPLATE_FILE_PATH);
+
+        if (job_token == null) {
+            job_token = DEFAULT_JOB_TOKEN;
+        }
+        if (script == null) {
+            script = DEFAULT_SCRIPT;
+        }
+
+        jobData = injectScriptAndTokenIntoXmlString(jobData, job_token, script);
+        return jobData;
+    }
+
+    /**
+     * Creates a jenkins job with a given name using the XML data as input for
+     * job configuration
+     *
+     * @param jobName
+     * @param jobXmlData
+     * @return
+     * @throws URISyntaxException
+     * @throws ClientProtocolException
+     * @throws IOException
+     */
+    public boolean createJob(String jobName, String jobXmlData) throws URISyntaxException {
+        HttpRequest httpRequest = new HttpRequest(HttpMethod.POST);
+        boolean success = false;
+
+        if (jobName == null) {
+            jobName = DEFAULT_JOB_NAME;
+        }
+
+        httpRequest.setJenkinsBaseUrl(jenkinsBaseUrl).addHeader("Authorization", "Basic " + encoding)
+                .addHeader("Content-type", MediaType.APPLICATION_XML).addHeader("Jenkins-Crumb", crumb)
+                .addParam("name", jobName).setBody(jobXmlData).setEndpoint("/createItem");
+
+        ResponseEntity response = httpRequest.performRequest();
+        success = response.getStatusCode() == HttpStatus.SC_OK;
+
+        if(!success) {
+            LOGGER.error("Failed to create a jenkins job. Status code: " + response.getStatusCodeValue() + ". Reason: " + response.getBody());
```

Comment:
Maybe this should also throw an error if it did not succeed? Then the user may or may not handle it depending on their implementation